### PR TITLE
test: More testing of restrictable enum declarations

### PR DESCRIPTION
**Pull request by JonathanStarup**

### CODE REVIEW

**Comment by magnus-madsen** [comment]
What about systematic tests for functions with signatures?

File: main/test/flix/Test.Dec.RestrictableTag.flix [hunk 1]
```diff
@@ -0,0 +1,71 @@
+mod Test.Dec.RestrictableTag {
+
+/// Simple nullary cases
```
Comment (mlutze) [inline main/test/flix/Test.Dec.RestrictableTag.flix:3]: No indent because we're waiting on the top-level namespace magic, right? :)

**Review by mlutze — APPROVED** [review]

**Comment by JonathanStarup** [comment]
> What about systematic tests for functions with signatures?

I thought I'd just have this in its own PR since its isolated

**Comment by magnus-madsen** [comment]
> > What about systematic tests for functions with signatures?
> 
> I thought I'd just have this in its own PR since its isolated

OK, so more tests to come.

**Pull request merged**